### PR TITLE
test: add TestTargetsFromGroupWithLabelKeepDrop to reinforce a relabelling behaviour

### PR DESCRIPTION
**Pull request by machine424**
fixes https://github.com/prometheus/prometheus/issues/12355

### CODE REVIEW

File: scrape/target_test.go [hunk 1]
```diff
@@ -354,6 +354,80 @@ func TestTargetsFromGroup(t *testing.T) {
...
+		name             string
+		cfgText          string
+		targets          []model.LabelSet
+		shouldDropTarget bool
+	}{
+		{
+			name: "no relabeling",
+			cfgText: `
+global:
+  metric_name_validation_scheme: legacy
```
Comment (machine424) [inline scrape/target_test.go:371]: Had to add those to get around:

```
2024/12/03 17:19:06 http: TLS handshake error from 127.0.0.1:52972: remote error: tls: bad certificate
--- FAIL: TestTargetsFromGroupWithLabelKeepDrop (0.00s)
    --- FAIL: TestTargetsFromGroupWithLabelKeepDrop/no_relabeling (0.00s)
panic: utf8 name validation requested but model.NameValidationScheme is not set to UTF8 [recovered]
	panic: utf8 name validation requested but model.NameValidationScheme is not set to UTF8 [recovered]
	panic: utf8 name validation requested but model.NameValidationScheme is not set to UTF8
```

(will look into that later, maybe we need to override a global var in this test file or sth...)

Comment (machine424) [reply scrape/target_test.go:371]: opened https://github.com/prometheus/prometheus/issues/15530 as this is not caused/related to the change in the PR.

File: scrape/target_test.go [hunk 2]
```diff
@@ -354,6 +354,80 @@ func TestTargetsFromGroup(t *testing.T) {
 	require.EqualError(t, failures[0], expectedError)
 }
 
+// TestTargetsFromGroupWithLabelKeepDrop aims to demonstrate and reinforce the current behavior: relabeling's "labelkeep" and "labeldrop"
+// are applied to all labels of a target, including internal ones (labels starting with "__" such as "__address__").
+// This will be hlpful for cases like https://github.com/prometheus/prometheus/issues/12355.
```
Comment (beorn7) [inline scrape/target_test.go:359]:
```suggestion
// This will be helpful for cases like https://github.com/prometheus/prometheus/issues/12355.
```

**Review by beorn7 — APPROVED** [review]
With tiny spelling nit. Feel free to merge after fixing it.

**Comment by machine424** [comment]
opened https://github.com/prometheus/prometheus/issues/15538 for the unrelated data race in tests.

File: scrape/target_test.go (hunk 1 above)
Comment (beorn7) [reply scrape/target_test.go:371]: Thanks for opening #15530 .

**Comment by beorn7** [comment]
And thanks for filing #15538

**Review by beorn7 — APPROVED** [review]

**Pull request merged**